src/pbt/v2/project/components/airflow_jobs.py

```diff
@@ -0,0 +1,443 @@
+                abs_file_path = os.path.join(root, file)
+                zipf.write(abs_file_path, abs_file_path[len(folder_path):])
+
+
+class AirflowJobDataProcessor:
+
+    def __init__(self, job_pbt: dict, prophecy_job_yaml: str, rdc: Dict[str, str], sha: Optional[str]):
+        self.job_pbt = job_pbt
+        self.prophecy_job_yaml = prophecy_job_yaml
+        self.rdc = rdc
+        self.sha = sha
+        self.prophecy_job_json_dict = {}
+
+        self._initialize_prophecy_job_json()
+
+    @property
+    def is_valid_job(self):
+        prophecy_job_yaml_dict = self.prophecy_job_json_dict
+        return self.job_pbt is not None and self.prophecy_job_yaml is not None and self.rdc is not None and \
+            prophecy_job_yaml_dict.get('metainfo', {}).get('fabricId', None) is not None
+
+    def validate_prophecy_managed_checksum(self, salt: str):
+        file_joiner: str = "$$$"
+        checksum = calculate_checksum(
+            file_joiner.join(file_content for file_content in filter_job_files(self.rdc).values()), salt)
+        return self.sha == checksum
+
+    @property
+    def job_files(self):
+        return filter_job_files(self.rdc)
+
+    @property
+    def dag_name(self):
+        return self.prophecy_job_json_dict.get('metainfo', {}).get('dagName', None)
+
+    @property
+    def is_disabled(self):
+        return self.prophecy_job_json_dict.get('enabled', False) is False
+
+    @property
+    def is_enabled(self):
+        return self.prophecy_job_json_dict.get('enabled', False) is True
+
+    @property
+    def fabric_id(self):
+        fabric_id = self.job_pbt.get(FABRIC_UID)
+        return str(fabric_id) if fabric_id is not None else None
+
+    @property
+    def pipelines(self):
+        return self.job_pbt.get('pipelines', [])
+
+    @property
+    def has_dbt_component(self):
+        return any(value.get('component', None) == 'dbt' for value in self.prophecy_job_json_dict['processes'].values())
+
+    def _initialize_prophecy_job_json(self):
+        try:
+            self.prophecy_job_json_dict = yaml.unsafe_load(self.prophecy_job_yaml)
+        except Exception as e:
+            print(f"Error while loading prophecy job yaml", e)
+            self.prophecy_job_json_dict = {}
+
+
+class AirflowJobs:
+    def __init__(self, project_parser: ProjectParser, project_config: ProjectConfig):
+        self._project_parser = project_parser
+        self._project_config = project_config
+
+        self._airflow_jobs: Dict[str, AirflowJobDataProcessor] = {}
+        self._in_valid_airflow_jobs: Dict[str, AirflowJobDataProcessor] = {}
+        self._airflow_jobs_without_code = {}
+
+        self.valid_airflow_jobs: Dict[str, AirflowJobDataProcessor] = {}
+        self.prophecy_managed_dbt_jobs: Dict[str, AirflowJobDataProcessor] = {}
+
+        self._airflow_clients = {}
+
+        self._initialize_airflow_jobs()
+        self._initialize_valid_airflow_jobs()
+        self._initialize_prophecy_managed_dbt_jobs()
```

Comment:
1. can we follow the pattern of assigning the instance-variable in the constructor itself, instead of delegating to another method? e.g. self.jobs = self._initialize_airflow_jobs()